File: kyu 7/complementaryDNA.js
const DNAStrand = (text) => {
  const arr = text.split("");
  let answer = [];
  arr.map((e) =>
    e == "A"
      ? answer.push("T")
      : e == "T"
      ? answer.push("A")
      : e == "C"
      ? answer.push("G")
      : e == "G"
      ? answer.push("C")
      : "nothing"
  );
  return answer.join("");
};

console.log(DNA_strand("AATCAA"));

function DNAStrand(dna) {
  return dna.replace(/./g, function (c) {
    return DNAStrand.pairs[c];
  });
}

DNAStrand.pairs = {
  A: "T",
  T: "A",
  C: "G",
  G: "C",
};

File: kyu 7/complementaryDNA.py
def DNA_strand(dna):
    answer = ''
    for d in dna:
        if d == 'A':
            answer += 'T'
        elif d == 'T':
            answer += 'A'
        elif d == 'C':
            answer += 'G'
        elif d == 'G':
            answer += 'C'
    return answer


print(DNA_strand("AAAA"))
